### ISSUE TITLE
Handle broken manifest gracefully

### ISSUE THREAD

**Issue by JensN4** [NumberFour/n4js#278]
Currently, a broken manifest causes the builder to throw NPEs and terminate. Consequently, no validation happens and no error markers are created. 

The NPE:

```
java.lang.NullPointerException
    at eu.numberfour.n4js.internal.N4JSModel.getProvidedRuntimeLibraries(N4JSModel.java:362)
    at eu.numberfour.n4js.internal.N4JSProject.getProvidedRuntimeLibraries(N4JSProject.java:203)
    at eu.numberfour.n4js.ui.internal.N4MFProjectDependencyStrategy.getProjectDependencies(N4MFProjectDependencyStrategy.java:49)
    at eu.numberfour.n4js.ui.internal.ProjectDescriptionLoadListener.getProjectDependenciesAsSet(ProjectDescriptionLoadListener.java:136)
    at eu.numberfour.n4js.ui.internal.ProjectDescriptionLoadListener.updateProjectReferencesIfNecessary(ProjectDescriptionLoadListener.java:101)
    at eu.numberfour.n4js.ui.building.N4JSBuildTypeTrackingBuilder.updateProjectReferencesIfNecessary(N4JSBuildTypeTrackingBuilder.java:93)
    at eu.numberfour.n4js.ui.building.N4JSBuildTypeTrackingBuilder.doBuild(N4JSBuildTypeTrackingBuilder.java:75)
    at org.eclipse.xtext.builder.impl.XtextBuilder.fullBuild(XtextBuilder.java:319)
    at org.eclipse.xtext.builder.impl.XtextBuilder.build(XtextBuilder.java:155)
    at org.eclipse.core.internal.events.BuildManager$2.run(BuildManager.java:735)
    at org.eclipse.core.runtime.SafeRunner.run(SafeRunner.java:42)
    at org.eclipse.core.internal.events.BuildManager.basicBuild(BuildManager.java:206)
    at org.eclipse.core.internal.events.BuildManager.basicBuild(BuildManager.java:246)
    at org.eclipse.core.internal.events.BuildManager$1.run(BuildManager.java:301)
    at org.eclipse.core.runtime.SafeRunner.run(SafeRunner.java:42)
    at org.eclipse.core.internal.events.BuildManager.basicBuild(BuildManager.java:304)
    at org.eclipse.core.internal.events.BuildManager.basicBuildLoop(BuildManager.java:360)
    at org.eclipse.core.internal.events.BuildManager.build(BuildManager.java:383)
    at org.eclipse.core.internal.events.AutoBuildJob.doBuild(AutoBuildJob.java:144)
    at org.eclipse.core.internal.events.AutoBuildJob.run(AutoBuildJob.java:235)
    at org.eclipse.core.internal.jobs.Worker.run(Worker.java:55)
```

Example for a broken manifest:

```
ArtifactId: n4js.example.tasks
VendorId: eu.numberfour
ProjectName: "n4js.example.tasks"
VendorName: "NumberFour AG"
ProjectType: library
ProjectVersion: 0.0.1
Output: "src-gen"
Sources {
    source {
        "src"
    }
}
RequiredRuntimeLibraries {
    n4js-runtime-es2015
}
ProjectDependencies {
    n4js.lang,
    mongodb,
    express
}
```

(Created issue from internal bug report by Moritz)

**Comment by lbeurerkellner** [comment]
I couldn't reproduce this bug. Looking at the code it became clear that @kduske-n4 must have fixed this in #259 by introducing null checks (75d423806c03c8d6e9f1f33e66a7c0c898a4c480).

**Comment by kduske-n4** [comment]
Couldn't reproduce either. The given example yields a parse error at `ArtefactId` as expected.
